src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -352,16 +353,20 @@ public void RemoveSubscriber_Impl(GuidId subscriptionId, StreamId streamId)
                     int maxCacheAddCount = queueCache?.GetMaxAddCount() ?? QueueAdapterConstants.UNLIMITED_GET_QUEUE_MSG;
                     if (maxCacheAddCount != QueueAdapterConstants.UNLIMITED_GET_QUEUE_MSG && maxCacheAddCount <= 0)
                         return;
-
+                    
+                    var retryMax = 6;
```

Comment:
retry max should probably be a const defined with the other static retry definitions (see ReadLoopBackoff)